src/time.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import moment from 'moment'
+
+var Time = React.createClass({
+  displayName: 'Time',
+
+  propTypes: {
+    selected: React.PropTypes.string,
+    onTimeClick: React.PropTypes.func
+  },
+  getDefaultProps () {
+    return {
+      selected: '12:00'
+    }
+  },
+
+  handleTimeClick (time) {
+    console.log(time);
+    // let t = times[i];
+    this.props.onTimeClick(time);
+  },
+
+  renderTimes () {
+    let times = [];
+    let startOfDay = moment().startOf('day');
+    let endOfDay = moment().endOf('day');
+    let time = startOfDay;
+
+    while (time <= endOfDay) {
+      times.push(time.format('h:mm a'));
+      time = time.clone().add(30, 'minutes');
+    }
```

Comment:
completely optional but this is basically a for loop:

```js
for (let time = startOfDay; time <= endOfDay; time = time.clone().add(30, 'minutes') {
  times.push(time.format('h:mm a'));
}
```


Side node: does `time <= endOfDay` do what you want? I think you may need to use something like `time.isBefore(endOfDay)`?